sklearn/utils/tests/test_sparsefuncs.py

```diff
@@ -478,3 +481,28 @@ def test_csc_row_median():
 
     # Test that it raises an Error for non-csc matrices.
     assert_raises(TypeError, csc_median_axis_0, sp.csr_matrix(X))
+
+
+def test_inplace_normalize_l1():
+    ones = np.ones((10, 1))
+    rs = RandomState(10)
+
+    for dtype in [np.float64, np.float32]:
+        X = rs.rand(10, 5).astype(dtype)
+        X_csr = sp.csr_matrix(X)
+        inplace_csr_row_normalize_l1(X_csr)
+        assert_equal(X_csr.dtype, dtype)
```

Comment:
Hello @MechCoder ,
I think test here already check that the dtype of X is preserved no matter it is `np.float32` or `np.float64`?